sentry/src/main/java/io/sentry/Hub.java

```diff
@@ -699,21 +699,12 @@ public void flush(long timeoutMillis) {
     boolean samplingDecision = tracingSampler.sample(samplingContext);
     transactionContexts.setSampled(samplingDecision);
 
-    SentryTransaction transaction = null;
-    if (!isEnabled()) {
-      options
-          .getLogger()
-          .log(
-              SentryLevel.WARNING,
-              "Instance is disabled and this 'startTransaction' call is a no-op.");
+    SentryTransaction transaction = new SentryTransaction(transactionContexts, this);
+    final StackItem item = stack.peek();
+    if (item != null) {
+      item.scope.setTransaction(transaction);
     } else {
-      final StackItem item = stack.peek();
-      if (item != null) {
-        transaction = new SentryTransaction(transactionContexts, this);
-        item.scope.setTransaction(transaction);
-      } else {
-        options.getLogger().log(SentryLevel.FATAL, "Stack peek was null when startTransaction");
-      }
+      options.getLogger().log(SentryLevel.FATAL, "Stack peek was null when startTransaction");
```

Comment:
not sure if it makes sense, SDK should be enabled in order to do its operations, and right now we don't check it anymore, what's the purpose of it?

Reply:
SDK can be disabled only after it is closed. When SDK is closed, check if it is closed will happen on transaction send in `captureTransaction` - so we still have these checks present.

Without this change, since `startTransaction` can return `null`, every time users call `startTransaction` they have to check if it's null and the code gets very verbose. While in Java you can simply ignore the fact that it can return null, imagine translating this code to Kotlin:

```java
SentryTransaction transaction = Sentry.startTransaction("processOrderBatch()");
transaction.setOperation("task");
try {
    procesOrderBatch();
} catch (Exception e) {
    transaction.setThrowable(e);
    transaction.setStatus(SpanStatus.INTERNAL_ERROR)
} finally {
    transaction.finish();
}
```

Reply:
it should return `NotNull`, that's for sure, and ideally, `NoOpTransaction` as just @bruno-garcia said.

true about being checked already in `captureTransaction`, agreed with the removal, I'd pass already a `NoOpTransaction` though.